docs.md

```diff
@@ -0,0 +1,372 @@
+the other nodes in your business network that need to be aware of the new accounts.
+
+### How do we mix workflows with accounts and non-accounts?
+
+This is a bit tricky for now. You'll need to update your flows with logic to determine if a state participant/owner is
+an account holder, or not. This can be done using the `AccountService` APIs. If so, then some additional steps should be
+taken to share the required `AccountInfo`s and `PublicKey` mappings with other nodes that need to see them. 
+
+Working with accounts is much like working with confidential identities but with the addition of the `AccountInfo` 
+state. For example, when using confidential identities, a new `PublicKey` would be requested from a counterparty node 
+before creating a transaction with that counterparty. The same workflow is required with accounts - a new `PublicKey` 
+should be requested before creating a state and transaction. The biggest difference is that, in your flow, you must also
+specify from which account the new `PublicKey` should be allocated to. However, of course, this should be known up-front
+by the node invoking the flow.
+
+## Are states for different accounts segregated?
+
+No. They are all stored in the same `VaultService` operated by the host node. States held by different accounts on the 
+same node can be partitioned by their account ID when querying the vault. See below for more info.
+
+## How do we handle permission on a per account basis?
+
+This must be done at the application level for now and can be handled in the RPC client part of your CorDapp.
+   
+## Typical workflows
+
+### Creating new accounts
+
+A new account can be created by either calling the `CreateAccount` flow via RPC or from an existing flow, 
+or by calling `AccountService.createAccount`. The `CreateAccount` method simply calls the `CreateAccount` flow and 
+returns a `CordaFuture` which completes when the `CreateAccount` flow successfully terminates.
+
+To create an account, you must specify an account name. the account ID is chosen randomly by the Accounts CorDapp. When
+the `CreateAccount` flow terminates, it returns a `StateAndRef<AccountInfo>` for the account which has just been 
+created. 
+
+The `CreateAccount` flow does not communicate with any nodes as it performs an internal process only (a notary signature
+is not required for state creation when the creating transaction does not include a `TimeWindow`).
+
+Code sample:
+
+    // Create account by invoking flow via RPC.
+    val accountInfo: StateAndRef<AccountInfo> = rpcProxy.startFlow(::CreateAccount, "Roger's account").returnValue.getOrThrow()
+    
+    // Create account by using sub flow (from inside a flow).
+    val accountInfo: StateAndRef<AccountInfo> = subFlow(CreateAccount("Roger's account"))
+    
+    // Create account using AccountService (from inside a flow). "accountService" is an extention function on FlowLogic.
```

Comment:
The later phrase is a little confusing (perhaps just remove altogether):
_"accountService" is an extention function on FlowLogic._

The AccountService provides access to triggering Account related flows from within a Corda Service.

Reply:
Thanks - will amend!